src/main/java/de/tum/cit/aet/artemis/quiz/dto/exercise/QuizExerciseForCourseDTO.java

```diff
@@ -0,0 +1,129 @@
+        if (quizExercise.getQuizBatches() != null) {
+            batches = quizExercise.getQuizBatches().stream().map(QuizBatchForCourseDTO::of).collect(Collectors.toSet());
+        }
+        return new QuizExerciseForCourseDTO(quizExercise.getId(), quizExercise.getTitle(), quizExercise.isQuizStarted(), quizExercise.isQuizEnded(), isEditable,
+                quizExercise.getDuration(), quizExercise.getMaxPoints(), quizExercise.getReleaseDate(), quizExercise.getStartDate(), quizExercise.getDueDate(),
+                quizExercise.getIncludedInOverallScore(), batches, quizExercise.getQuizMode(), quizExercise.isIsOpenForPractice());
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        QuizExerciseForCourseDTO that = (QuizExerciseForCourseDTO) o;
+        if (id != that.id) {
+            return false;
+        }
+        if (quizStarted != that.quizStarted) {
+            return false;
+        }
+        if (quizEnded != that.quizEnded) {
+            return false;
+        }
+        if (isEditable != that.isEditable) {
+            return false;
+        }
+        if (duration != that.duration) {
+            return false;
+        }
+        if (Double.compare(that.maxPoints, maxPoints) != 0) {
+            return false;
+        }
+        if (!title.equals(that.title)) {
+            return false;
+        }
+        if (!Objects.equals(releaseDate, that.releaseDate)) {
+            return false;
+        }
+        if (!Objects.equals(startDate, that.startDate)) {
+            return false;
+        }
+        if (!Objects.equals(dueDate, that.dueDate)) {
+            return false;
+        }
+        if (!includedInOverallScore.equals(that.includedInOverallScore)) {
+            return false;
+        }
+        if (quizMode != that.quizMode) {
+            return false;
+        }
+        if (!Objects.equals(isOpenForPractice, that.isOpenForPractice)) {
+            return false;
+        }
+        boolean thisBatchesEmpty = quizBatches == null || quizBatches.isEmpty();
+        boolean thatBatchesEmpty = that.quizBatches == null || that.quizBatches.isEmpty();
+        if (thisBatchesEmpty && thatBatchesEmpty) {
+            return true;
+        }
+        if (thisBatchesEmpty != thatBatchesEmpty) {
+            return false;
+        }
+        return Objects.equals(quizBatches, that.quizBatches);
+    }
+
+}
+
+@JsonInclude(JsonInclude.Include.NON_EMPTY)
+record QuizBatchForCourseDTO(long id, @Nullable String password, boolean started, boolean ended) {
+
+    public static QuizBatchForCourseDTO of(QuizBatch quizBatch) {
+        return new QuizBatchForCourseDTO(quizBatch.getId(), quizBatch.getPassword(), quizBatch.isStarted(), quizBatch.isEnded());
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
I don't think we need to save chars here 😄 
```suggestion
    public boolean equals(Object object) {
```